mobile/lib/ui/actions/collection/collection_file_actions.dart

```diff
@@ -84,6 +84,103 @@ extension CollectionFileActions on CollectionActions {
     }
   }
 
+  Future<bool> addToMultipleCollections(
+    BuildContext context,
+    List<Collection> collections,
+    bool showProgressDialog, {
+    List<EnteFile>? selectedFiles,
+  }) async {
+    final List errors = [];
+    final ProgressDialog? dialog = showProgressDialog
+        ? createProgressDialog(
+            context,
+            S.of(context).uploadingFilesToAlbum,
+            isDismissible: true,
+          )
+        : null;
+    await dialog?.show();
+    try {
+      for (final collection in collections) {
+        try {
+          final List<EnteFile> files = [];
+          final List<EnteFile> filesPendingUpload = [];
+          final int currentUserID = Configuration.instance.getUserID()!;
+          for (final file in selectedFiles!) {
+            EnteFile? currentFile;
+            if (file.uploadedFileID != null) {
+              currentFile = file.copyWith();
+            } else if (file.generatedID != null) {
+              // when file is not uploaded, refresh the state from the db to
+              // ensure we have latest upload status for given file before
+              // queueing it up as pending upload
+              currentFile = await (FilesDB.instance.getFile(file.generatedID!));
+            } else if (file.generatedID == null) {
+              logger.severe("generated id should not be null");
+            }
+            if (currentFile == null) {
+              logger.severe("Failed to find fileBy genID");
+              continue;
+            }
+
+            if (currentFile.uploadedFileID == null) {
+              currentFile.collectionID = collection.id;
```

Comment:
Was this line accidentally omitted? See `addToCollection`:
https://github.com/ente-io/ente/blob/60246be8611405b4d26231e500321673c6addc00/mobile/lib/ui/actions/collection/collection_file_actions.dart#L140